### ISSUE TITLE
Data encryption is broken in mido

### ISSUE THREAD

**Issue by jacopotediosi** [PixelExperience/android-issues#449]
## Build date

PixelExperience_Plus_mido-11.0-20210720-1114-OFFICIAL

## Expected Behavior

Phone should encrypt and then reboot

## Current Behavior

Bootloop

## Steps to Reproduce

After a clean install (unencrypted device):
1.go to settings,
2.then security,
3.at the bottom "Encryption and identifiers"
4.press "encrypt the phone"
5.follow the process until the phone displays bugdroid, reboot and goes to bootloop

/device Mido
/version eleven_plus

**Comment by pixelexperiencebot** [comment]
Issue created! You can close at any time by commenting ```/close```

**Comment by GiaSen** [comment]
Looks like it encrypts and boots fine here, but there are random reboots, I just logged them and I'm going to check what makes it to die randomly

**Comment by jacopotediosi** [comment]
My device went into bootloop without even reaching the screen indicating the remaining time for encryption. I could hear the classic camera "clicks" of continuous reboots.
Unfortunately I cannot send logs, because after a couple of attempts (even with the penultimate PE 11+) I have downgraded to PE 10+ (which encrypts fine) and I have already reinstalled all my data on it.

**Comment by GiaSen** [comment]
No problem, I've alrady grabbed a log

**Comment by Zedasix** [comment]
Me to on redmi note 8 pro

**Comment by timplaul** [comment]
Same behavior with Motorola Moto G6 (ali) PixelExperience_ali-11.0-20210730-1709

**Comment by GiaSen** [comment]
You have to open another issue for different device

**Comment by jeangraff30** [comment]
> Mesmo comportamento com Motorola Moto G6 (ali) PixelExperience_ali-11.0-20210730-1709

This device is no longer supported by PE

**Comment by kprastey** [comment]
Build = PixelExperience_Plus_mido-11.0-20210720-1114-OFFICIAL
Device = Xiaomi Redmi Note 4 Global (Snapdragon)

I tried modifying the fstab.qcom file and my mido was able to forceencrypt on boot and it didn't go into bootloop.

**!Caution! try the below process only if you know what you're doing.**
- Firstly I had disabled the DM-parity and ForceEncryption on my device(optional).
- Flash the "PixelExperience_Plus_mido-11.0-20210720-1114-OFFICIAL" image like you ususlly do(don't exit the recovery yet).
- mount the system and vendor partitions from your custom recovery.
- Connect your phone to a computer that has adb drivers and fetch the "fstab.qcom" file using the `adb pull /vendor/etc/fstab.qcom` command.
- open the pulled file and change the flag `encryptable` to `forceencrypt` on the /data partition flags(SDHC is optional).
- Now push the "fstab.qcom" file using the `adb push fstab.qcom /vendor/etc/fstab.qcom` command.
- Now boot to system like you normally do after an image flash.

The device now should take some time to boot so wait for it and your mido should now boot as encrypted.

@jacopotediosi @jeangraff30 @GiaSen 

p.s. => Loved to have back the Dirac support. If this Issue is fixed in the next build PE community surely deserves a donation.
